lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -100,19 +113,23 @@ func (s *subscriberdbServicer) SyncSubscribers(
 		return nil, err
 	}
 	toRenew, deleted := subscriberdb.GetPerSubscriberDigestsDiff(clientPerSubDigests, cloudPerSubDigests)
-	if len(toRenew) > s.changesetSizeTheshold {
+	if len(toRenew) > s.changesetSizeThreshold || len(toRenew) > int(s.maxProtosLoadSize) {
 		return &lte_protos.SyncSubscribersResponse{Resync: true}, nil
 	}
+
 	sids := funk.Keys(toRenew).([]string)
-	subProtosById, err := subscriberdb.LoadSubProtosByID(sids, networkID, apnsByName, apnResourcesByAPN)
+	subProtosToRenew, err := s.subStore.GetSubscribers(networkID, sids)
 	if err != nil {
 		return nil, err
 	}
+	// Since the cached sub protos don't contain gateway-specific apn resources data, need to
+	// append that to each sub proto in this handler
+	subProtosToRenew = appendApnResourcesToSubProtos(subProtosToRenew, apnResourcesByAPN)
 
 	res := &lte_protos.SyncSubscribersResponse{
 		FlatDigest:    &lte_protos.Digest{Md5Base64Digest: flatDigest},
 		PerSubDigests: cloudPerSubDigests,
-		ToRenew:       subProtosById,
+		ToRenew:       subProtosToRenew,
```

Comment:
"subProtosToRenew" -> "renewed", because that var actually does contain the renewed values, not just a list of which subs should be renewed